compiler/optimizer/LoopReplicator.cpp

```diff
@@ -60,6 +60,23 @@ TR_LoopReplicator::TR_LoopReplicator(TR::OptimizationManager *manager)
    _maxNestingDepth = 0;
    }
 
+
+//Add static debug counter for a given replication failure
+static void countReplicationFailure(TR::Compilation *comp, char *failureReason, int32_t regionNum)
+   {
+   //Assemble format string: "LoopReplicator/<failureReason>/%s/(%s)/region_%d"
+   char *formatPre = "LoopReplicator/";
+   char *formatPost = "/%s/(%s)/region_%d";
+   //Make the buffer sufficiently large.
+   int32_t formatStringLength = strlen(formatPre) + strlen(failureReason) + strlen(formatPost) + 20;
+   char formatString[formatStringLength];
+   strcpy(formatString, formatPre);
+   strcat(formatString, failureReason);
+   strcat(formatString, formatPost);
```

Comment:
Rather than put `failureReason` into the format string, it would be better to fill it in `printf`-style into `LoopReplicator/%s/%s/(%s)/region_%d`